java/vector/src/main/java/org/apache/arrow/vector/dictionary/DictionaryProvider.java

```diff
@@ -79,15 +84,24 @@ public final Set<Long> getDictionaryIds() {
     }
 
     @Override
-    public Dictionary lookup(long id) {
+    public BaseDictionary lookup(long id) {
       return map.get(id);
     }
 
     @Override
     public void close() {
-      for (Dictionary dictionary : map.values()) {
+      for (BaseDictionary dictionary : map.values()) {
```

Comment:
Does the dictionary itself need to close()?

Reply:
Good question, what model of ownership does the Java impl follow? In the batch dictionary I offered a means of passing in two vectors in which case I'd assume the creator would be responsible for closing. Maybe I should remove that and if folks need nested dictionaries later we offer a ctor with a parent vector? Then the dictionary could be closed here.

Reply:
It makes sense to let the creator close. It should be documented though. The Java codebase is somewhat inconsistent about ownership unfortunately.

Reply:
Ack, thanks.